include/seastar/core/future.hh

```diff
@@ -1354,12 +1354,15 @@ public:
     /// Gets the value returned by the computation.
     ///
     /// Similar to \ref get(), but instead of returning a
-    /// tuple, returns the first value of the tuple.  This is
-    /// useful for the common case of a \c future<T> with exactly
-    /// one type parameter.
+    /// \c T&&, returns \c T.
     ///
-    /// Equivalent to: \c std::get<0>(f.get()).
+    /// \note Under most circumstances, it is equivalent to:
+    /// \c std::get<0>(f.get()). But when performing meta programming,
```

Comment:
I'm not sure what you're saying here. In "most circumstances", the future's type T won't be a tuple at all, so `get<0>(f.get())` won't even work. I'm guessing this example worked in the past, but not today?

Maybe it's better to more explicitly say something like the following: (it's just something I quickly wrote, feel free to completely reword):

The get0() method is deprecated, left over from earlier versions of Scylla that had variadic futures - futures that returned multiple values. Back then, get0() was useful for getting the first (and usually, only) value. Today just get() does the same thing, although because of the different return type (get0() returned T, get() returns T&&) some metaprogramming will need to use (... insert details here). In new code using a future<tuple<...>>, use `std::get<0>(f.get())` instead of `f.get0()` to get the first component of the tuple.